Localization for Lua

Edru2 opened the pull request:
#### Brief overview of PR changes/additions
This PR gives the possibility to translate the right click menu of adjustable containers
The method is also universally usable for other parts of the lua code.
It uses a default file in translations/lua/mudlet-lua.json which contains the default (english) text.
This file is then usable by Crowdin. The translated files will be in translations/lua/translated.

Added function: 
**loadTranslations**(packageName, [fileName], [languageCode], [folder])
only the packageName is not optional. 
filename if not given is "mudlet-lua"
languageCode if not given is taken from mudlet.translations.interfacelanguage
folder if not given will is either "translations/lua/" or "translations/lua/translated"

A packagename would be for example "AdjustableContainer"
Inside the function the translations could be loaded like:
myTranslations = loadTranslations("AdjustableContainer")

The Json file is organized like a https://support.crowdin.com/file-formats/chrome-json/ to allow support of description and message

#### Motivation for adding to Mudlet
As Adjustable Containers are now part of Mudlet they also need translations (at least the right click menu)

#### Other info (issues closed, discussion etc)
I tried my best on my Italian and German translation but someone else should also look over it. ( They are more a showcase for what is translatable and how it would work)

If someone has better ideas how to make localization or suggestion for the translation I put in already (also for the English ones of course ;) ) please leave feedback but be aware that the space of the right click label is a bit limited  (I could make it bigger if really needed, but try to avoid that)

Edru2 committed:
first commit

Allow localization of Adjustable Containers right click Menu.

vadi2 commented:
The idea is good! But let's see if we can translate [through Crowdin](https://wiki.mudlet.org/w/Translating_Mudlet) to keep translations in 1 easy place for our translators.

We have two options...

1) do it via C++ - put strings into https://github.com/Mudlet/Mudlet/blob/development/src/TLuaInterpreter.cpp#L16973 and they'll show up in Crowdin and be usable in Lua via `mudlet.translate`. Not super nice because it means we have to edit C++ code to be able to use a string in Lua
2) Setup something custom to import/export from Crowdin, it supports many formats including csv and txt: https://support.crowdin.com/supported-formats/

I think 2 is worth investigating to have a general solution for getting Lua strings into Crowdin, what do you think? Would you be able to look at it?

Edru2 commented:
Will have a look.
This will be even more important for https://github.com/Mudlet/MudletPackageManager

Edru2 committed:
added translations

translations has the default translations as json file
translated has the finished translation for them as json files

if file in translated cannot be found the default file will be loaded.
The default file can be used also in crowdin as default file.
with the settings /translated/%locale_with_underscore%_%original_file_name% in the file settings

demonnic commented:
Agreed, the i18n lua module is better than nothing, but it would be nice if we could utilize the same resources.

Edru2 commented:
hmm. my yesterdays commit never arrived. (github had probably some issues)
 Edit: I was wrong. I see it here now :man_facepalming: 
I had an approach with json files loading (which are useable by crowdin)
It was working but had still a few flaws.



> Agreed, the i18n lua module is better than nothing, but it would be nice if we could utilize the same resources.

I was talking about the client side. Didn't look into the server side yet.
If there is a better solution and I'm able to somehow integrate the i18n module (didn't look how it works) it will even be better.

Edit: Ah I see now that it arrived

vadi2 commented:
Oh nice, I will try to load this into Crowdin.

vadi2 committed:
Format & sort file

vadi2 commented:
Could you add context to the strings? "light" for example can mean many things in a language, so we should make it less ambiguous for translators.

![Selection_557](https://user-images.githubusercontent.com/110988/79952717-1ec08a00-847b-11ea-9a8a-cda6da3a38a4.png)

vadi2 commented:
Here are the strings in Crowdin: https://crowdin.com/translate/mudlet/180/en-de

vadi2 commented:
I translated it for Russian then downloaded it:

[AdjustableContainer.zip](https://github.com/Mudlet/Mudlet/files/4514451/AdjustableContainer.zip)

Seems OK? If yes then it's something we can automate like the `.ts` translations.

Edru2 commented:
It should be okay. the file name needs to follow some rules.
you can change it in crowdin into this /translated/%locale_with_underscore%_%original_file_name%

vadi2 commented:
Yes. 

What about putting the files into the same root `/translations` folder, so we don't have them in many locations?

Edru2 commented:
> What about putting the files into the same root `/translations` folder, so we don't have them in many locations?

Had the same idea.
That was one of the flaws I was talking about.
Need to check how the lua script can reach the root folder.

vadi2 commented:
Using `../` in paths? Though there are two problems: 1) loading when running Mudlet in a development environment and 2) loading when Mudlet is running as an installed version. Let's solve 1) first

Edru2 committed:
delete translations in lua and put the to the standard path

Edru2 commented:
> Using `../` in paths? Though there are two problems: 1) loading when running Mudlet in a development environment and 2) loading when Mudlet is running as an installed version. Let's solve 1) first

Okay. That was easier then expected.
Maybe still create it's own folder in translation so I doesn't litter all over the .ts files?

Now I'll have look if we chan change to use chrome.json files.

vadi2 commented:
Own folder sounds good

Edru2 committed:
change to chrome.json format and put everything it's own folder

Also added the russion translation vadi provided
(plz check if there is no copy/past error ;D)

Edru2 commented:
Done. Now everything should have context as well.
To see the context you maybe have to delete and re-add the file in crowdin.

vadi2 commented:
OK. I'll look at setting up Crowdin integration for it. Could you merge latest `development` meanwhile so we can start looking at how a built version would be like?

vadi2 commented:
OK, I believe https://github.com/Mudlet/Mudlet/pull/3664 will do the job once this is in development.

Will wait on others to review this.

vadi2 commented:
`translations/lua_translations/AdjustableContainer/translated/` seems a bit redundant. What about `translations/translated-AdjustableContainer/`?

Edru2 committed:
lang code at the end and less redundant foldername

Edru2 committed:
Merge branch 'development' into Adjustable.container.localization

Edru2 commented:
> Okay. That was easier then expected.

I was wrong :wink:
It's not working. That it worked for me was just coincidence because I started Mudlet from right directory.

There are probably some changes in mudlet.pro needed to include the files but I have never played with it (besides copy pasting things I knew what they do)

And even if that is done. I have no idea how the script can find the translations directory then and if the path will be different for different OS.

I'll try my luck but I'll probably need some help.

vadi2 commented:
Yep happy to help with it

Edru2 committed:
find directory approach

Edru2 committed:
Update Other.lua

Edru2 committed:
add folder to check for appimage

vadi2 committed:
Build using patched version of installers

Edru2 committed:
another try

Edru2 commented:
it should work now (at least for linux)

vadi2 commented:
I can confirm it works on Linux.

Kebap commented on CI/travis.linux.after_success.sh:
```diff
@@ -27,7 +27,7 @@ if [ "${DEPLOY}" = "deploy" ]; then
   fi
 
-  git clone https://github.com/Mudlet/installers.git "${TRAVIS_BUILD_DIR}/../installers"
+  git clone https://github.com/Edru2/installers.git -b patch-1 "${TRAVIS_BUILD_DIR}/../installers"
```
Don't we want to link Mudlet git in release?

vadi2 replied:
Yes! This is just to test https://github.com/Mudlet/installers/pull/76.

vadi2 replied:
@Edru2 need to revert this now.

Edru2 replied:
Just be aware that Linux builds won't have the translation files till https://github.com/Mudlet/installers/pull/76 is not merged.

vadi2 replied:
Merged now.

Kebap commented:
May I suggest reworking the names of files & folders to be even more concise like so:

Established already:
* Input: `/translations/mudlet.ts`
* Output: `/translations/translated/mudlet_pl_PL.ts`

Current proposal:
* Input: `/translations/lua_translations/AdjustableContainer.json`
* Output: `/translations/lua_translations/AdjustableContainer-translated/AdjustableContainer_ru_RU.json`

My suggestion:
* Input: `/translations/lua/AdjustableContainer.json`
* Output: `/translations/lua/translated/AdjustableContainer_ru_RU.json`

I am not sure about using individual files for different lua components.
Currently we combine all other Mudlet strings in a single file.
Maybe your approach has its advantages though.

We need to review how these Input files & folders will be shown in Crowdin as well:
![](https://wiki.mudlet.org/images/e/e8/Mudlet-translation-file.png)

vadi2 commented:
The new folder structure sounds good to me, but combining different formats would be a pain.

Kebap commented:
I am not suggesting to combine mudlet.ts with AdjustableContainer.json at all.
I was merely thinking about other lua texts that may need l10n later.
Would they result in additional lua files with their names?
Or will we combine them in a mudlet-lua.json?

vadi2 commented:
Sorry, I misunderstood. Yeah, I think that would be feasible!

Edru2 commented:
Put them all in one mudlet-lua.json would mean that, to be organized, the json file would have to look like 
```
{
  "AdjustableContainer":{
    "attach": { 
      "message": "Attach to:",
      "description":"Attaches your container to a border"
    },  
    "bottom": { 
      "message":"bottom",
      "description":"is on of the possible borders to attach the containers"
    },
...
```
That means that it the json file is not recognized as chrome.json by Crowdin and so the description (context) for every single item would have to be put on Crowdin manually.

So, it would be possible. Just bigger effort :wink:
And I think that it is more comfortable to see the filename on crowdin directly to know what the translation is for.

Both ways are okay for me.

Edru2 committed:
change to only one file for all lua translations

Kebap commented:
Sorry for brainstorming during your development time here.
I see you have already put the above thoughts into code. 

It's really more effort to add descriptions manually on Crowdin. 
Especially if the file changes and needs to be updated again.
They should be all available in the file from the beginning.
This way the coders can describe their texts immediately.
If Crowdin can't handle that, the files should stay seperated.
Unless there is some way to make it work with the descriptions?

If they are seperated, we just need to think about the implications a little bit more at some time.
Until now, we just tell the translators to open `mudlet.ts` and they will find "all" Mudlet strings.
In the future, they will also have to select `AdjustableContainer.json`, otherwise it stays English.
When there come more json files for different lua modules, they need to open each one.
Also not sure if the 95% calculation for the language golden stars need adjustments.
Then again, not sure if this is the perfect time for these considerations right here.

Edru2 committed:
description also possible now with one file

Edru2 commented:
So, with a bit of hackery I was able to get everything in one file (with description working as well)
It should also work on windows now.

vadi2 commented:
I can confirm it's working on Linux and Windows now 👍

Edru2 committed:
clean up and error messages added

vadi2 commented on src/mudlet-lua/lua/Other.lua:
```diff
@@ -942,41 +942,50 @@ function translateTable(data, language)
+      outputTable[k:gsub("^.*%.", "")] = inputTable[k]
+    end
+  end
+  if not(next(outputTable)) then
```
```suggestion
  if table.is_empty(outputTable) then
```

vadi2 commented:
The Sparkle error is strange and unrelated. I've re-ran the build, maybe it's transient.

```
2020-04-26T08:57:57.3718820Z ##[group]Run lukka/run-cmake@v2
2020-04-26T08:57:57.3719570Z with:
2020-04-26T08:57:57.3719910Z   cmakeListsOrSettingsJson: CMakeListsTxtAdvanced
2020-04-26T08:57:57.3720290Z   cmakeListsTxtPath: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/CMakeLists.txt
2020-04-26T08:57:57.3720600Z   useVcpkgToolchainFile: true
2020-04-26T08:57:57.3721020Z   buildDirectory: /Users/runner/runners/2.169.1/work/Mudlet/b/ninja
2020-04-26T08:57:57.3721430Z   cmakeAppendedArgs: -G Ninja -DCMAKE_PREFIX_PATH=/Users/runner/runners/2.169.1/work/Mudlet/Qt/5.14.1/clang_64/
2020-04-26T08:57:57.3723350Z   cmakeSettingsJsonPath: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet//CMakeSettings.json
2020-04-26T08:57:57.3723660Z   cmakeBuildType: Debug
2020-04-26T08:57:57.3724050Z   cmakeGenerator: Ninja
2020-04-26T08:57:57.3724130Z   configurationRegexFilter: .*
2020-04-26T08:57:57.3724430Z   buildWithCMake: true
2020-04-26T08:57:57.3724720Z   useShell: true
2020-04-26T08:57:57.3725020Z env:
2020-04-26T08:57:57.3725490Z   qt: 5.14.1
2020-04-26T08:57:57.3726660Z   vcpkgResponseFile: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/our-vcpkg-dependencies/vcpkg-x64-osx-dependencies
2020-04-26T08:57:57.3727150Z   RUNVCPKG_VCPKG_ROOT: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg
2020-04-26T08:57:57.3727670Z   VCPKG_ROOT: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg
2020-04-26T08:57:57.3728120Z   AZP_CACHING_CONTENT_FORMAT: Files
2020-04-26T08:57:57.3728440Z   RUNVCPKG_VCPKG_TRIPLET: x64-osx
2020-04-26T08:57:57.3728890Z   RUNVCPKG_VCPKG_TRIPLET_OUT: x64-osx
2020-04-26T08:57:57.3729090Z ##[endgroup]
2020-04-26T08:57:58.8684490Z tool: /usr/local/bin/cmake
2020-04-26T08:57:58.8952900Z tool: 
2020-04-26T08:57:58.9213740Z Downloading url 'https://github.com/ninja-build/ninja/releases/download/v1.10.0/ninja-mac.zip' to file '/Users/runner/runners/2.169.1/work/Mudlet/b/dl/https___github.com_ninja-build_ninja_releases_download_v1.10.0_ninja-mac.zip'.
2020-04-26T08:57:58.9494830Z 'https://github.com/ninja-build/ninja/releases/download/v1.10.0/ninja-mac.zip' downloaded to: '/Users/runner/runners/2.169.1/work/Mudlet/b/dl/https___github.com_ninja-build_ninja_releases_download_v1.10.0_ninja-mac.zip'
2020-04-26T08:57:58.9506040Z Extracting archive '/Users/runner/runners/2.169.1/work/Mudlet/b/dl/https___github.com_ninja-build_ninja_releases_download_v1.10.0_ninja-mac.zip' ...
2020-04-26T08:57:59.0357630Z ##[group]Generate project files with CMake
2020-04-26T08:58:04.4317230Z -- The C compiler identification is AppleClang 11.0.0.11000033
2020-04-26T08:58:05.0845000Z -- The CXX compiler identification is AppleClang 11.0.0.11000033
2020-04-26T08:58:05.3298220Z -- Check for working C compiler: /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
2020-04-26T08:58:06.6373710Z -- Check for working C compiler: /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - works
2020-04-26T08:58:06.6672960Z -- Detecting C compiler ABI info
2020-04-26T08:58:07.4718980Z -- Detecting C compiler ABI info - done
2020-04-26T08:58:07.5375180Z -- Detecting C compile features
2020-04-26T08:58:07.5488350Z -- Detecting C compile features - done
2020-04-26T08:58:07.6018100Z -- Check for working CXX compiler: /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
2020-04-26T08:58:08.4251330Z -- Check for working CXX compiler: /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - works
2020-04-26T08:58:08.4366950Z -- Detecting CXX compiler ABI info
2020-04-26T08:58:09.1728360Z -- Detecting CXX compiler ABI info - done
2020-04-26T08:58:09.2307050Z -- Detecting CXX compile features
2020-04-26T08:58:09.2311490Z -- Detecting CXX compile features - done
2020-04-26T08:58:09.2606210Z -- Including optional updater module
2020-04-26T08:58:09.2693250Z -- Including optional fonts module
2020-04-26T08:58:09.2745310Z -- Including optional 3D mapper module
2020-04-26T08:58:09.2870830Z -- Including optional build-type splash screen module
2020-04-26T08:58:09.3565050Z -- Sparkle CocoaPod is missing, running 'pod install' to get it...
2020-04-26T08:58:28.4573730Z Analyzing dependencies
2020-04-26T08:58:28.4576480Z Adding spec repo `trunk` with CDN `https://cdn.cocoapods.org/`
2020-04-26T08:58:28.4577060Z Downloading dependencies
2020-04-26T08:58:28.4577570Z Installing Sparkle (1.23.0)
2020-04-26T08:58:45.5836580Z checking for a BSD-compatible install... /usr/bin/install -c
2020-04-26T08:58:46.8132170Z checking whether build environment is sane... yes
2020-04-26T08:58:46.9275320Z checking for a thread-safe mkdir -p... ./install-sh -c -d
2020-04-26T08:58:46.9361600Z checking for gawk... no
2020-04-26T08:58:46.9460080Z checking for mawk... no
2020-04-26T08:58:46.9585570Z checking for nawk... no
2020-04-26T08:58:46.9652590Z checking for awk... awk
2020-04-26T08:58:50.8325610Z checking whether make sets $(MAKE)... yes
2020-04-26T08:58:50.8878440Z checking for gcc... /Applications/Xcode_11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
2020-04-26T08:58:51.2880770Z checking whether the C compiler works... yes
2020-04-26T08:58:51.2886290Z checking for C compiler default output file name... a.out
2020-04-26T08:58:51.5393370Z checking for suffix of executables... 
2020-04-26T08:58:51.7002190Z checking whether we are cross compiling... configure: error: in `/Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/edbee-lib/vendor/onig':
2020-04-26T08:58:51.7096860Z configure: error: cannot run C compiled programs.
2020-04-26T08:58:51.7116270Z If you meant to cross compile, use `--host'.
2020-04-26T08:58:51.7119310Z See `config.log' for more details
2020-04-26T08:58:51.8928780Z -- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
2020-04-26T08:58:52.2869690Z -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
2020-04-26T08:58:52.8402600Z -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
2020-04-26T08:58:52.8571450Z -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
2020-04-26T08:58:53.4442430Z -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
2020-04-26T08:58:53.4634340Z -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
2020-04-26T08:58:54.0504670Z -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
2020-04-26T08:58:54.3374340Z -- Building translations
2020-04-26T08:58:57.0132880Z warning: lua-yajl not available - translation statistics in settings won't be shown.
2020-04-26T08:58:57.6445900Z -- Qt version: 5.14.1
2020-04-26T08:58:57.6564610Z -- Using AppleClang compiler
2020-04-26T08:58:57.7005540Z -- Checking for one of the modules 'zip;libzip'
2020-04-26T08:58:59.3122250Z -- Found ZIP: /usr/local/Cellar/libzip/1.6.1/lib/libzip.dylib (found version "1.6.1") 
2020-04-26T08:58:59.4854460Z -- Found Lua51: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/debug/lib/liblua.a (found version "5.1.5") 
2020-04-26T08:58:59.7898060Z -- Found ZLIB: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/debug/lib/libz.a (found version "1.2.11") 
2020-04-26T08:58:59.8482850Z -- Checking for one of the modules 'pcre;libpcre'
2020-04-26T08:59:01.4802480Z -- Found PCRE: /usr/local/Cellar/pcre/8.44/lib/libpcre.dylib (found version "8.44") 
2020-04-26T08:59:01.5128980Z -- Checking for one of the modules 'yajl;libyajl'
2020-04-26T08:59:01.8092080Z -- Found YAJL: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/debug/lib/libyajl.dylib  
2020-04-26T08:59:01.8446980Z -- Checking for one of the modules 'pugixml;libpugixml'
2020-04-26T08:59:02.5728920Z CMake Warning (dev) at /usr/local/Cellar/cmake/3.17.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
2020-04-26T08:59:02.5732270Z   The package name passed to `find_package_handle_standard_args` (PugiXML)
2020-04-26T08:59:02.5733130Z   does not match the name of the calling package (PUGIXML).  This can lead to
2020-04-26T08:59:02.5733610Z   problems in calling code that expects `find_package` result variables
2020-04-26T08:59:02.5736880Z   (e.g., `_FOUND`) to follow a certain pattern.
2020-04-26T08:59:02.5737370Z Call Stack (most recent call first):
2020-04-26T08:59:02.5738020Z   cmake/FindPUGIXML.cmake:38 (find_package_handle_standard_args)
2020-04-26T08:59:02.5738480Z   3rdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:286 (_find_package)
2020-04-26T08:59:02.5738950Z   src/CMakeLists.txt:258 (find_package)
2020-04-26T08:59:02.5743330Z This warning is for project developers.  Use -Wno-dev to suppress it.
2020-04-26T08:59:02.5743670Z 
2020-04-26T08:59:02.5745660Z -- Found PugiXML: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/lib/libpugixml.a (found version "1.10") 
2020-04-26T08:59:02.6081810Z -- Checking for one of the modules 'hunspell;libhunspell'
2020-04-26T08:59:02.9096470Z -- Found HUNSPELL: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/debug/lib/liblibhunspell.a  
2020-04-26T08:59:04.1854110Z -- Found Boost: /Users/runner/runners/2.169.1/work/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-osx/include (found suitable version "1.72.0", minimum required is "1.44")  
2020-04-26T08:59:04.2677570Z -- Found OpenGL: /Applications/Xcode_11.3.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/OpenGL.framework   
2020-04-26T08:59:04.2796550Z -- Using TextToSpeech module
2020-04-26T08:59:04.2905410Z -- Using Gamepad module
2020-04-26T08:59:05.7308260Z CMake Warning (dev) at /usr/local/Cellar/cmake/3.17.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
2020-04-26T08:59:05.7315960Z   The package name passed to `find_package_handle_standard_args` (SPARKLE)
2020-04-26T08:59:05.7317300Z   does not match the name of the calling package (Sparkle).  This can lead to
2020-04-26T08:59:05.7318440Z   problems in calling code that expects `find_package` result variables
2020-04-26T08:59:05.7319000Z   (e.g., `_FOUND`) to follow a certain pattern.
2020-04-26T08:59:05.7319630Z Call Stack (most recent call first):
2020-04-26T08:59:05.7320230Z   cmake/FindSparkle.cmake:9 (find_package_handle_standard_args)
2020-04-26T08:59:05.7320740Z   3rdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:286 (_find_package)
2020-04-26T08:59:05.7349110Z   3rdparty/sparkle-glue/CMakeLists.txt:72 (find_package)
2020-04-26T08:59:05.7352000Z This warning is for project developers.  Use -Wno-dev to suppress it.
2020-04-26T08:59:05.7364360Z 
2020-04-26T08:59:05.7367580Z -- Could NOT find SPARKLE (missing: SPARKLE_INCLUDE_DIR) 
2020-04-26T08:59:05.7405680Z -- Configuring done
2020-04-26T08:59:06.5548690Z CMake Error at 3rdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:186 (_add_executable):
2020-04-26T08:59:06.5562180Z   Target "mudlet" links to target "Sparkle::Sparkle" but the target was not
2020-04-26T08:59:06.5562970Z   found.  Perhaps a find_package() call is missing for an IMPORTED target, or
2020-04-26T08:59:06.5563320Z   an ALIAS target is missing?
2020-04-26T08:59:06.5563620Z Call Stack (most recent call first):
2020-04-26T08:59:06.5564200Z   src/CMakeLists.txt:271 (add_executable)
2020-04-26T08:59:06.5564440Z 
2020-04-26T08:59:06.5564510Z 
2020-04-26T08:59:06.5564790Z CMake Error at 3rdparty/vcpkg/scripts/buildsystems/vcpkg.cmake:215 (_add_library):
2020-04-26T08:59:06.5570710Z   Target "cocoa-qt-glue" links to target "Sparkle::Sparkle" but the target
2020-04-26T08:59:06.5571020Z   was not found.  Perhaps a find_package() call is missing for an IMPORTED
2020-04-26T08:59:06.5571360Z   target, or an ALIAS target is missing?
2020-04-26T08:59:06.5571680Z Call Stack (most recent call first):
2020-04-26T08:59:06.5572580Z   3rdparty/sparkle-glue/CMakeLists.txt:65 (add_library)
2020-04-26T08:59:06.5572660Z 
2020-04-26T08:59:06.5572930Z 
2020-04-26T08:59:06.7591400Z -- Generating done
2020-04-26T08:59:06.7763370Z CMake Generate step failed.  Build files cannot be regenerated correctly.
2020-04-26T08:59:06.8088470Z ##[endgroup]
2020-04-26T08:59:06.8099430Z ##[error]Error: "CMake failed with error code: '1'.
2020-04-26T08:59:06.8140170Z ##[error]run-cmake action execution failed
```

Edru2 committed:
Update src/mudlet-lua/lua/Other.lua

Co-Authored-By: Vadim Peretokin <vperetokin@gmail.com>

Edru2 commented:
Can you give me some directions how to add the language files to the osx package (no idea where that happens).
After that's working it should be ready to review :smiley:

vadi2 commented:
Off the top of my memory - look around for a similar script as Linux in the generic-osx folder.

Edru2 committed:
attempt to add translation files to macx

Edru2 committed:
another go

Edru2 committed:
Update mudlet.pro

Edru2 committed:
Update mudlet.pro

vadi2 commented:
Nice, confirmed working in macOS as well.

Edru2 committed:
Revert "Build using patched version of installers"

This reverts commit 602cdea0960358a1ca5cd2fff20f128e0651f38c.

SlySven commented:
Can I point out that our putting the translations outside of the `./src`  directory does make things more stranger in Qt Creator (and the handling of resources files) - not that it doesn't work but it make things work harder... :stuck_out_tongue_winking_eye:

SlySven commented on src/mudlet-lua/lua/geyser/GeyserAdjustableContainer.lua:
```diff
@@ -810,15 +812,14 @@ function Adjustable.Container:new(cons,container)
-    me.attLabel.txt  = me.attLabel.txt or [[<font size="5" face="Noto Emoji">⚓</font> Attach to:]]
-    me.lockStylesLabel.txt = me.lockStylesLabel.txt or [[<font size="5" face="Noto Emoji">🖌</font> Lockstyle:]]
-    me.customItemsLabel.txt = me.customItemsLabel.txt or [[<font size="5" face="Noto Emoji">🖇</font> Custom:]]
+    me.lockLabel.txt = me.lockLabel.txt or [[<font size="5" face="Noto Emoji">🔒</font>]] .. self.Locale.lock.message
```
This hard encoded use of Noto Emoji is only valid for Linux builds - other OSes cannot (Windows 10)/do not (MacOs) use that font...

vadi2 replied:
That's OK - it tested and emoji shows up fine on all OSs. So it's harmless for others.

Edru2 replied:
> This hard encoded use of Noto Emoji is only valid for Linux builds - other OSes cannot (Windows 10)/do not (MacOs) use that font...

I'm aware of that.
This is just a workaround to always have emojis with colour in Linux.
It happens that they lost their colour if the adjustable container was created in a userwindow or a new Style-sheet was set.

SlySven commented on src/mudlet.pro:
```diff
@@ -761,6 +762,10 @@ LUA_GEYSER.files = \
 LUA_GEYSER.depends = mudlet
 
+LUA_TRANSLATIONS.files = \
+    $${PWD}/../translations/lua/*
```
:-1: You *cannot* (the last time I checked) wild-card the file names when making a **Unix** GNU Makefile in QMake so that it can do a `make install` - you have to list each file individually (see all the 250+ files that I had to put in to support the Lua Code Formatter :nauseated_face:) ...

vadi2 replied:
@Githzerai could you confirm this?

Edru2 replied:
make install is working as expected in Ubuntu 19.10
Don't know about other systems though.

vadi2 replied:
We don't need the 250 files listed explicitly in the `.pro` actually? @SlySven ?

SlySven replied:
You used to have to - and I have not seen anything to suggest that has changed...

vadi2 reviewed (approved):
Looks great :+1:

Kebap commented on translations/lua/mudlet-lua.json:
```diff
@@ -1,19 +1,62 @@
+  },
+  "AdjustableContainer.right": {
+    "message":"right",
+    "description":"one of the borders the container can be attached to"
```
Top, left and bottom all have the same description, only this one needs adjustment: 
```suggestion
    "description":"is one of the possible borders the container can be attached to"
```

Kebap commented on translations/lua/mudlet-lua.json:
```diff
@@ -1,19 +1,62 @@
+  },
+  "AdjustableContainer.load": {
+    "message":"Load",
+    "description":"Loads the previously settings of the container"
```
```suggestion
    "description":"Loads the previous settings of the container"
```

Edru2 replied:
hm I changed this one already. You are somehow referring to a older version.

the translations (all of them) were more a showcase how it works, I think they should be rechecked/rediscussed on Crowdin.

Kebap replied:
My bad. You are right!

Edru2 committed:
Update mudlet-lua.json

Kebap commented:
@vadi2 Do we need to prepare anything Crowdin for the files to be read and returned as expected, or can we just merge?

vadi2 commented:
Let's merge it first, and then I'll prepare the Crowdin integration.

Edru2 committed:
loadTranslations rewrite to check default

Had to rewrite loadTranslations, so it checks also in default files if it didn't find a translation in translated json
